man/systemd-cryptenroll.xml

```diff
@@ -235,6 +235,30 @@
     limitation does not apply to PKCS#11 tokens.</para>
   </refsect1>
 
+  <refsect1>
+    <title>Compatibility</title>
+
+    <para>Security technology both in systemd and in the general industry constantly evolves. In order to
+    provide best security guarantees, the way TPM2, FIDO2, PKCS#11 devices are enrolled is regularly updated
+    in newer versions of systemd. Whenever this happens the following compatibility guarantees are given:</para>
+
+    <itemizedlist>
+      <listitem><para>Old enrollments continue to be supported and may be unlocked with newer versions of
+      <citerefentry><refentrytitle>systemd-cryptsetup@.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>.</para></listitem>
+
+      <listitem><para>The opposite is not guaranteed however: it might not be possible to unlock volumes with
+      enrollments done with a newer version of <command>systemd-cryptenroll</command> with an older version
+      of <command>systemd-cryptsetup</command>.</para></listitem>
+    </itemizedlist>
+
+    <para>That said, it is generally recommended to use matching versions of
+    <command>systemd-cryptenroll</command> and <command>systemd-cryptsetup</command>, since this is best
+    tested and supported.</para>
+
+    <para>It might be advisable to re-enroll existing enrollments regularly to take benefit of newer security
```

Comment:
Constant re-enrollment is a pain that we want to get away from, so I'd drop this para

Reply:
the thing is that early tpm enrollments didn't make use of param encryption, pinning and salting. You only get those if you re-enroll. I'll reword it slightly suggesting to do it only after updates.

Reply:
Not just after any upgrade, but after upgrades that introduce new features that are desired